packages/ocaml-monadic/ocaml-monadic.0.3.1/opam

```diff
@@ -0,0 +1,26 @@
+opam-version: "1.2"
+maintainer: "JHU PL Lab <pl.cs@jhu.edu>"
+authors: "JHU PL Lab <pl.cs@jhu.edu>"
+homepage: "https://github.com/zepalmer/ocaml-monadic"
+bug-reports: "https://github.com/zepalmer/ocaml-monadic/issues"
+license: "BSD-3-clause"
+dev-repo: "https://github.com/zepalmer/ocaml-monadic.git"
+build: [
+  ["oasis" "setup"]
```

Comment:
On Fri, 2 Oct 2015 13:55:12 -0700, Zachary Palmer wrote:
> 
> I'm still a bit of a novice at this, so please forgive my questions. :) If I understand correctly, you're suggesting that setup.ml be included in the tarball so that the build and build-test in the repo just rely on setup.ml itself?

Yes, it was a design goal of oasis that you can distribute you software without requiring the user to install oasis itself.

> We added setup.ml to our .gitignore because we didn't want to put generated content into our repository. Is that not best practice in this scenario?

It is... for development.  For distribution, I suggest you generate a tarball and add it to your Github release.  A Make rule to generate such tarball is:


PKGNAME	    = $(shell oasis query name)
PKGVERSION  = $(shell oasis query version)
PKG_TARBALL = $(PKGNAME)-$(PKGVERSION).tar.gz

DISTFILES   = LICENSE.txt INSTALL.txt README.md ...

dist tar: setup.ml
	mkdir $(PKGNAME)-$(PKGVERSION)
	cp -r --parents $(DISTFILES) $(PKGNAME)-$(PKGVERSION)/
#	Generate a setup.ml independent of oasis
	cd $(PKGNAME)-$(PKGVERSION) && oasis setup
	tar -zcvf $(PKG_TARBALL) $(PKGNAME)-$(PKGVERSION)
	$(RM) -rf $(PKGNAME)-$(PKGVERSION)